[Elixir] Concept exercise: basics (#1118)
exercism/v3: languages/elixir/exercises/concept/basics/.docs/after.md

> Elixir is a dynamically-typed language, meaning that the type of a variable is only checked at run-time. There is also no real concept of variable assignment.  Using the [match `=` operator][match], we can bind a value to a variable name:

Reviewer: > There is also no real concept of variable assignment.  

What exactly do you mean by this?

Author: ```suggestion
Elixir is a dynamically-typed language, meaning that the type of a variable is only checked at run-time. In Elixir, variables are not assigned, instead they are bound to a value.  Using the [match `=` operator][match], we can bind a value to a variable name:
```

Author: I think I should change it to this

Reviewer: Ok sweet - that's a good improvement. 

So, I still don't know the difference, but in the introduction that's ok. However, I wonder about removing the sentence from the introduction altogether and moving it to the after.md and expanding more there what the difference between assigning and binding is. I don't feel like I need to know that nugget of information to be able to successfully solve the exercise so it could potentially be removed from the introduction.md. Thoughts?

Author: I can agree on that, i don't think it really matters at this point of learning because in elixir it acts a lot like assignment.

But if you say assignment to many elixirists (elixtrisians? I love making up words for people who write in a language :stuck_out_tongue:), it sits funny since in erlang you can't rebind a variable and the value is bound until the variable is out of scope. So it is an important distinction at some point.

Reviewer 2: It's related to how pattern matching works in the Erlang abstract machine. The = operator actually performs a pattern match. When pattern matches are successful, which can be in other ways than =,  all unbound variables on the left-hand-side are bound to whatever's on the right-hand-side. I think it's an important distinction, and the word "binding" should be used, but the explanation why could probably be saved until pattern matching is introduced.

(Sorry, coming in from Erlang, crashing your review!)

Author: @gwaerondor No worries! I really appreciate the input! Agreed, I think we should save the details of this until pattern matching is discussed.  I made some changes to the location of the this information to the `after.md` document, I think it still reflects this distinction but doesn't delve too deeply into it yet.

Reviewer 3: >  Agreed, I think we should save the details of this until pattern matching is discussed.

This seems wise.

> Elixir is a dynamically-typed language, meaning that the type of a variable is only checked at run-time.  In Elixir, we refer to variable assignement as being _bound_ to a value.  Using the [match `=` operator][match], we can bind a value to a variable name:

Reviewer 3: ```suggestion
Elixir is a dynamically-typed language, meaning that the type of a variable is only checked at run-time.  In Elixir, we refer to variable assignment as being _bound_ to a value.  Using the [match `=` operator][match], we can bind a value to a variable name:
```

Author: My fingers let me down. [cue sad charlie brown music]
